packages/builders/src/components/Components.ts

```diff
@@ -15,14 +15,15 @@ export interface MappedComponentTypes {
 export function createComponent<T extends keyof MappedComponentTypes>(
 	data: APIMessageComponent & { type: T },
 ): MappedComponentTypes[T];
-export function createComponent(data: APIMessageComponent): Component {
+export function createComponent<C extends MessageComponent>(data: C): C;
+export function createComponent(data: APIMessageComponent | MessageComponent): Component {
```

Comment:
Shouldn't you just.. type it as `data: APIMessageComponent | Component`?

Reply:
can't
![image](https://user-images.githubusercontent.com/38259440/151070233-a33b15aa-432f-44a1-9c78-77cdb9bf0cc7.png)